test/test_response.py

```diff
@@ -165,6 +165,57 @@ def test_chunked_decoding_gzip(self):
         assert r.read() == b''
         assert r.read() == b''
 
+    def test_decode_gzip_multi_member(self):
+        import zlib
+        compress = zlib.compressobj(6, zlib.DEFLATED, 16 + zlib.MAX_WBITS)
+        data = compress.compress(b'foo')
+        data += compress.flush()
+        data = data * 3
+
+        fp = BytesIO(data)
+        r = HTTPResponse(fp, headers={'content-encoding': 'gzip'})
+
+        assert r.data == b'foofoofoo'
+
+    def test_decode_gzip_error(self):
+        import zlib
+        compress = zlib.compressobj(6, zlib.DEFLATED, 16 + zlib.MAX_WBITS)
+        data = b'foo'
+
+        fp = BytesIO(data)
+        with pytest.raises(DecodeError):
+            HTTPResponse(fp, headers={'content-encoding': 'gzip'})
+
+    def test_decode_gzip_swallow_garbage(self):
+        import zlib
+        compress = zlib.compressobj(6, zlib.DEFLATED, 16 + zlib.MAX_WBITS)
+        data = compress.compress(b'foo')
+        data += compress.flush()
+        data = data * 3 + b'foo'
+
+        fp = BytesIO(data)
+        r = HTTPResponse(fp, headers={'content-encoding': 'gzip'},
+                         preload_content=False)
+        ret = b''
+        for _ in range(100):
+            ret += r.read(1)
+            if r.closed:
+                break
+
+        assert ret == b'foofoofoo'
+
+    def test_chunked_decoding_gzip_swallow_garbage(self):
```

Comment:
What do you mean by `chunked` here? I usually associate that with `Transfer-Encoding: chunked` so a docstring on this test would be helpful

Reply:
This is the same terminology in the deflate tests I think.  The difference is that here the decoding is done with preload_content=False, and after that read() with various lengths.  Here the main objective is to test the case where the read's continues after the first failure, which require such "chunked" decoding to be exercised.

Reply:
@sigmavirus24, does that answer your question?

Reply:
👍